backend/src/apis/ownership.rs

```diff
@@ -0,0 +1,37 @@
+use crate::contracts::{generated::summa_contract::AddressOwnershipProof, signer::SummaSigner};
+use ethers::types::Address;
+use std::{error::Error, result::Result};
+
+pub struct AddressOwnership {
+    address_ownership_proofs: Vec<AddressOwnershipProof>,
+    signer: SummaSigner,
+}
+
+impl AddressOwnership {
+    pub fn new(
+        main_signer_key: &str,
+        chain_id: u64,
+        rpc_url: &str,
+        address: Address,
+    ) -> Result<AddressOwnership, Box<dyn Error>> {
+        Ok(AddressOwnership {
+            address_ownership_proofs: vec![],
+            signer: SummaSigner::new(&[], main_signer_key, chain_id, rpc_url, address),
+        })
+    }
+
+    // Make sure the input `addresses` in `address_ownership_proof` are not duplicated with addresses already registered on the contract.
+    pub async fn dispatch_proof_of_address_ownership(
+        &mut self,
+        address_ownership_proof: Vec<AddressOwnershipProof>,
```

Comment:
Based on the current implementation, we expect the Exchange to build these `AddressOwnershipProof` by themselves without providing any helpers for doing that.

So, according to the previous suggestion, the function doesn't have to take address_ownership_proof as input anymore as there are already created in `new`.

Therefore, the only role of this function is to dispatch the proof to the contract

Reply:
Updated comment at [ce908f0](https://github.com/summa-dev/summa-solvency/pull/152/commits/ce908f0a7323153aea8f3f959f16e69766a661d7)